### PR TITLE
EOS-26640: Add SupportBundle Filter Interfaces.

### PR DESCRIPTION
**Pull request by gargpalak048**
# Problem Statement
- Provide interfaces for filters in SB Framework for components to use.

# Design
-  Added a class FiletrSupportBundle under SB Framework to mock filter interfaces

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: 
-  Confirm All CODACY errors are resolved [Y/N]: 

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [ ] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [ ] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [ ] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [x] Changes done to WIKI / Confluence page
https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/749076684/Node+level+Support+Bundle+Collection

### CODE REVIEW

File: py-utils/src/utils/support_framework/sb_filters.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+class FilterSupportBundle:
+    """Provides Filter interfaces for support bundle."""
+
+    @staticmethod
+    def limit_size(src_dir, file_name_reg_ex, size, dest_dir):
```
Comment (sachinpunadikar) [inline py-utils/src/utils/support_framework/sb_filters.py:23]: Copy the input file to the dest_dir as it is

File: py-utils/src/utils/support_framework/sb_filters.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        msg = "limit_size filter is not implemented yet"
+        return msg
+
+    @staticmethod
+    def limit_time(start_time_and_duration, src_dir, file_name_reg_ex, dest_dir):
```
Comment (sachinpunadikar) [inline py-utils/src/utils/support_framework/sb_filters.py:28]: Copy the input file to the dest_dir as it is

File: py-utils/src/utils/support_framework/sb_filters.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        msg = "limit_time filter is not implemented yet"
+        return msg
```
Comment (sachinpunadikar) [inline py-utils/src/utils/support_framework/sb_filters.py:30]: Log this message in SB log file, do not return it.

File: py-utils/src/utils/support_framework/sb_filters.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+class FilterSupportBundle:
+    """Provides Filter interfaces for support bundle."""
+
+    @staticmethod
+    def limit_size(src_dir, file_name_reg_ex, size, dest_dir):
+        msg = "limit_size filter is not implemented yet"
+        return msg
```
Comment (sachinpunadikar) [inline py-utils/src/utils/support_framework/sb_filters.py:25]: Log this message in SB log file, do not return it.

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
work on comments

File: py-utils/src/utils/support_framework/sb_filters.py (hunk 1 above)
Comment (gargpalak048) [reply py-utils/src/utils/support_framework/sb_filters.py:27]: Addressed

File: py-utils/src/utils/support_framework/sb_filters.py (hunk 2 above)
Comment (gargpalak048) [reply py-utils/src/utils/support_framework/sb_filters.py:36]: Addressed

File: py-utils/src/utils/support_framework/sb_filters.py (hunk 3 above)
Comment (gargpalak048) [reply py-utils/src/utils/support_framework/sb_filters.py]: removed the return statement

File: py-utils/src/utils/support_framework/sb_filters.py (hunk 4 above)
Comment (gargpalak048) [reply py-utils/src/utils/support_framework/sb_filters.py]: removed the return statement

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/src/utils/support_framework/sb_filters.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import os
+import shutil
+
+
+class FilterLog:
```
Comment (cdeshmukh) [inline py-utils/src/utils/support_framework/sb_filters.py:23]: i think 'Logs' can be the class name, filters are of member functions acting on those logs

File: py-utils/src/utils/support_framework/sb_filters.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
```
Comment (sachinpunadikar) [inline py-utils/src/utils/support_framework/sb_filters.py:1]: file name -> log_filters.py

File: py-utils/src/utils/support_framework/sb_filters.py (hunk 5 above)
Comment (sachinpunadikar) [reply py-utils/src/utils/support_framework/sb_filters.py:23]: In my opinion the current class-name is correct as we are handing filtering only. 
"Logs" may indicate we are handing everything related to logs, so current name is good.

File: py-utils/src/utils/support_framework/sb_filters.py (hunk 6 above)
Comment (gargpalak048) [reply py-utils/src/utils/support_framework/sb_filters.py]: done.

**Review by rahul-tripathi-git — CHANGES_REQUESTED** [review]
Please refer: https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/749076684/Node+level+Support+Bundle+Collection#Future-implementations-(PI-5)
and make the needed changes.

File: py-utils/src/utils/support_framework/sb_filters.py (hunk 5 above)
Comment (cdeshmukh) [reply py-utils/src/utils/support_framework/sb_filters.py]: > "Logs" may indicate we are handing everything related to logs,

Not necessarily right, we will publish what is supported for class or even code can be checked.
Ideally class name needs to be noun, and any action on it as member functions.

Comment (cdeshmukh) [reply py-utils/src/utils/support_framework/sb_filters.py]: anyways, in interest of mock interfaces availability, am to merge this PR and address class name during actual implementation

**Review by cdeshmukh — APPROVED** [review]

**Comment by gargpalak048** [comment]
> Please refer: https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/749076684/Node+level+Support+Bundle+Collection#Future-implementations-(PI-5) and make the needed changes.

Changed the sequence of args for filter interfaces.

**Review by rahul-tripathi-git — APPROVED** [review]

File: py-utils/src/utils/support_framework/sb_filters.py (hunk 5 above)
Comment (rahul-tripathi-git) [reply py-utils/src/utils/support_framework/sb_filters.py]: Resolving this conversation.

**Pull request merged**